### ISSUE TITLE
Characters can't be Converted

### ISSUE THREAD

**Issue by mechPenSketch** [coppolaemilio/dialogic#1510]
## The problem

**Describe the bug**
The character files do not appear in the designated folder even when the conversion is completed.

It seems that `ResourceSaver.save(current_character, current_character.resource_path)` returns 15, which represents file unrecognized error.

**To Reproduce**
1. Go to Dialogic > Settings > Dialogic 1 Converter
2. Click "Verify Files"
3. Click "Begin Conversions"

**Expected behavior**
For at least a character file to be generated: `converted-dialogic/characters/example.dch`

**Screenshots**
![dialogic](https://user-images.githubusercontent.com/15977859/231195626-942578d3-5199-48b3-ad3c-4cb198c82240.png)

**System (please complete the following information):**
 - Windows 10
 - Godot 4.0.1
 - Dialogic Version: 2.0-Alpha-6 w/ further modifications:
   - https://github.com/coppolaemilio/dialogic/pull/1509 and #1505 implemented
   - I have also corrected a filepath in `settings_converter.gd` line 788 to:
```
current_character.resource_path = dir_char + folderPath + fileName + ".dch"
```

## Solutions

**Workaround**

N/A

**Possible fixes**

I believe this happens because `ResorceSaver` cannot save files that are not in `.res`, `.scn`, or their text-based counterparts.
* A new `ResourceFormatSaver` could be added to let it save in `.dch`.
* If it's text-based, I could rewrite the code to make it write the files similarly to the timelines.
* Save character in `.res` instead.

**Comment by mechPenSketch** [comment]
Since `ResorceSaver` is a singleton, `add_resource_format_saver(...)` should be called just once. I probably could do it inside `plugin.gd`'s `_enable_plugin()`.

As for adding resource format saver, `ResourceFormatSaver` could be extended, where `_get_recognized_extensions(_r)` returns `[".dch"]`.

**Comment by metanoder** [comment]
@mechPenSketch 

Hi.  I have a pr #1495 that found a few minor spelling errors in the original converter thebardsrc created.  I wonder if you wanna take a spin with that pr file.  The characters converted and timelines got converted into DTL when I ran my game through it.

**Comment by mechPenSketch** [comment]
@metanoder where exactly are your characters converted to?

**Comment by metanoder** [comment]
Where?  In folders created by the converter.  For characters, it would be a subfolder within converted-dialogic, 'characters'.  

<img width="699" alt="image" src="https://user-images.githubusercontent.com/117393840/232531485-81cd80e4-ad05-4f0e-b283-82286d7612dc.png">

**Comment by mechPenSketch** [comment]
I've tried your pr, but the characters still can't be converted on my side. How did your character conversion work when mine doesn't?

Judging by your screenshot, are you using an apple computer? Maybe the conversion works depending on the OS?

I'll try your PR later again to confirm.

**Comment by metanoder** [comment]
@mechPenSketch 

tldr; the conversation system works on the pc as well. 

I have access to a pc as well.  I downloaded a fresh copy of Godot 3.5.2, 4.0.2, Dialogic 1.4.5, and my pr Dialogic 2 alpha 6.  

I made a fresh project in 3.5.2 called psuedo game, and made three characters, and three timelines and backgrounds with a simple play through.   

Long story short, thebardsrc conversation system worked on the pc as well. 

![Screenshot 2023-04-18 194330](https://user-images.githubusercontent.com/117393840/232961693-2833abca-574e-477e-b943-0736ea1beee7.png)

![Screenshot 2023-04-18 194443](https://user-images.githubusercontent.com/117393840/232961716-7d3d0e40-7622-4ebf-a227-a1fb3d68028b.png)

![characters](https://user-images.githubusercontent.com/117393840/232961807-ecf13cc2-9554-4161-a97f-1ea9be11f735.jpg)

![Screenshot 2023-04-18 194511](https://user-images.githubusercontent.com/117393840/232961876-de8aa30c-d4ea-4911-9c88-80ce4226ed2c.png)

![Screenshot 2023-04-18 194548](https://user-images.githubusercontent.com/117393840/232961889-22270fb2-969e-4a5e-87d7-4238a2b6707c.png)

**Comment by mechPenSketch** [comment]
I've tried your PR again. I need to do a little bit of tweaking, but after that, the conversion surprisingly works, especially for the characters:
![dialogic](https://user-images.githubusercontent.com/15977859/233048174-b0c73e8a-89a1-4691-aa75-802e549dbb63.png)
I don't know how it worked, but once your PR is accepted, the conversion is no longer broken. I'm closing this issue.
